src/server/lsHost.ts

```diff
@@ -126,7 +126,18 @@ namespace ts.server {
         }
 
         resolveModuleNames(moduleNames: string[], containingFile: string): ResolvedModule[] {
-            return this.resolveNamesWithLocalCache(moduleNames, containingFile, this.resolvedModuleNames, this.resolveModuleName, m => m.resolvedModule);
+            return this.resolveNamesWithLocalCache(moduleNames, containingFile, this.resolvedModuleNames, this.resolveModuleName, m => {
+                if (!m.resolvedModule) {

```

Comment:
it will be better to do the following:
- add a constraint `R extends { resolvedFileName?: string }` to `R` type parameter in `resolveNamesWithLocalCache` - both `ResolvedModule` and `ResolvedTypeReferenceDirective` satisfy this constraint.
- inside `moduleResolutionIsValid` so something like:
```ts
// lastDeletedFileName should be declared as somewhere before the `for-of` loop 
// const lastDeletedFileName = this.project.projectService.lastDeletedFile && this.project.projectService.lastDeletedFile.fileName;

const result = getResult(resolution);
if (result) {
    if (result.resolvedFileName && result.resolvedFileName === lastDeletedFileName) {
        return false;
    }
    return true;
}
```